app/models/user.rb

```diff
@@ -670,6 +670,16 @@ def update_visit_record!(date)
     create_visit_record!(date) unless visit_record_for(date)
   end
 
+  def update_timezone_if_missing(timezone)
+    return if timezone.blank?
+
+    # we only want to update the user's timezone if they have not set it themselves
+    UserOption
```

Comment:
There's no need to select the timezone before updating it here. You could do this:

`UserOption.where(user_id: self.id, timezone: nil).update_all(timezone: timezone)`

It will be one SQL query instead of two.

Reply:
Ah thank you for that, was trying to golf it to the smallest query possible. Will change